File: backend/src/controllers/pacienteController.js
import { PacienteRepository } from '../repositories/pacienteRepository.js';
import { ConsultaRepository } from '../repositories/consultaRepository.js';

const pacienteRepo = new PacienteRepository();
const consultaRepo = new ConsultaRepository();

export async function postCadastroPaciente(req, res, next) {
  try {
    const out = await pacienteRepo.upsert(req.body || {});
    res.status(201).json(out);
  } catch (e) {
    next(e);
  }
}

export async function patchPacienteObservacoes(req, res, next) {
  try {
    const pacienteId = req.body?.cpf || req.body?.pacienteId || req.body?.telefone;
    const observacoes = req.body?.observacoes;
    const out = await pacienteRepo.updateObservacoes(pacienteId, observacoes);
    res.json(out);
  } catch (e) {
    next(e);
  }
}

export async function deletePaciente(req, res, next) {
  try {
    const paciente = await pacienteRepo.getByCpf(req.params.id);
    await consultaRepo.deleteByPatient(req.params.id, paciente?.telefone);
    res.json(await pacienteRepo.delete(req.params.id));
  } catch (e) { next(e); }
}

File: backend-java/src/main/java/com/clinica/bot/controller/PacienteController.java
package com.clinica.bot.controller;

import com.clinica.bot.domain.Paciente;
import com.clinica.bot.repository.ConsultaRepository;
import com.clinica.bot.repository.PacienteRepository;
import com.clinica.bot.security.SecurityUtils;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequiredArgsConstructor
public class PacienteController {
    private final PacienteRepository pacientes;
    private final ConsultaRepository consultas;

    @PostMapping("/pacientes/cadastro")
    @ResponseStatus(HttpStatus.CREATED)
    public Paciente cadastrar(@RequestBody Map<String, Object> body) {
        return pacientes.upsert(body);
    }

    @PatchMapping("/pacientes/observacoes")
    public Paciente observacoes(@RequestBody Map<String, Object> body) {
        Object id = body.get("cpf") != null ? body.get("cpf")
                : body.get("pacienteId") != null ? body.get("pacienteId") : body.get("telefone");
        return pacientes.updateObservacoes(id == null ? "" : String.valueOf(id),
                body.get("observacoes") == null ? "" : String.valueOf(body.get("observacoes")));
    }

    @DeleteMapping("/pacientes/{id}")
    public Map<String, Object> excluir(@PathVariable String id) {
        SecurityUtils.requireUser();
        Paciente paciente = pacientes.getByCpf(id).orElse(null);
        consultas.deleteByPatient(id, paciente == null ? null : paciente.getTelefone());
        return pacientes.delete(id);
    }
}
